flowapi/flowapi/permissions.py

```diff
@@ -267,83 +138,61 @@ def schema_to_scopes(schema: dict) -> Iterable[str]:
     >>> list(schema_to_scopes({"FlowmachineQuerySchema": {"oneOf": [{"$ref": "DUMMY"}]},"DUMMY": {"properties": {"query_kind": {"enum": ["dummy"]}}},},))
     ["get_result&dummy", "run&dummy", "get_result&available_dates"],
     """
-    yield from per_query_scopes(
-        queries=ResolvingParser(spec_string=dumps(schema)).specification["components"][
-            "schemas"
-        ]["FlowmachineQuerySchema"]
-    )
-
-
-def expand_scopes(*, scopes: List[str]) -> str:
-    """
-    Expand up a list of compact scopes to full scopes
-
-    Parameters
-    ----------
-    scopes : list of str
-        Compressed scopes to expand
 
-    Yields
-    ------
-    str
-        A scope string
-
-    """
-    for scope in scopes:
-        parts = scope.strip().split("&")
-        ps = (x.split(",") for x in parts)
-        yield from (set(x) for x in product(*ps))
+    # Note from meeting; this will need to be per-role check, as all permissions for a query have to be contained in
+    # a single role
+
+    # Example query scopes:
+    # "run",
+    #  "read",
+    #  "spatial_aggregate",
+    #  "locations:admin_1",
+    #  "locations:admin_3",
+    #  "event_dates:1990-02-01:1992-03-04"
+    #  "event_type:mds",
+    #  "event_type:sms",
+    #  "subscriber_subset"
+
+    # Boolean permissions:
+    # Check run
+    # Check read
+    # Check subscriber subset
+    # Check event types
+    # Check query tree
+    # Check dates
+    resolved_queries = ResolvingParser(spec_string=dumps(schema)).specification[
+        "components"
+    ]["schemas"]["FlowmachineQuerySchema"]
+    unique_scopes = []
+    for tl_query in resolved_queries["oneOf"]:
+        tl_query_name = tl_query["properties"]["query_kind"]["enum"][0]
+        print(f"Looking for {tl_query_name}")
+        query_list = grab_on_key_list(
+            tl_query,
+            ["properties", "query_kind", "enum", 0],
+        )
+        if query_list == []:
+            return []
```

Comment:
Some queries don't have any sub-queries, but we'll still need scopes for them (e.g. `<agg_unit>:<tl_query_name>:`)

Reply:
Resolved: all queries now have the TL query as a sub_query (ex: `admin1:tl_query:tl_query`)